code/modules/mob/living/stats.dm

```diff
@@ -65,357 +46,172 @@
-				if(STAEND > 10)
-					heart.sellprice = STAEND*2
-					if(STAEND > 12)
-						heart.sellprice = STAEND*3
-						if(STAEND > 14)
-							heart.sellprice = STAEND*5
-							if(STAEND > 16)
-								heart.sellprice += STAEND*7
-
-			var/obj/item/organ/lungs/lungs = getorganslot(ORGAN_SLOT_LUNGS)
-			if(lungs)
-				if(STAEND <= 10)
-					lungs.sellprice = STAEND
-				if(STAEND > 10)
-					lungs.sellprice = STAEND*2
-					if(STAEND > 12)
-						lungs.sellprice = STAEND*3
-						if(STAEND > 14)
-							lungs.sellprice = STAEND*5
-							if(STAEND > 16)
-								lungs.sellprice += STAEND*7	Commented out since no gating to fix goblins etc being farmed for truly ludicrous amounts*/
-
-		if("speed")
-			newamt = STASPD + amt
-			if(BUFSPE < 0)
-				BUFSPE = BUFSPE + amt
-				if(BUFSPE > 0)
-					newamt = STASPD + BUFSPE
-					BUFSPE = 0
-			if(BUFSPE > 0)
-				BUFSPE = BUFSPE + amt
-				if(BUFSPE < 0)
-					newamt = STASPD + BUFSPE
-					BUFSPE = 0
-			while(newamt < 1)
-				newamt++
-				BUFSPE--
-			while(newamt > 20)
-				newamt--
-				BUFSPE++
-			STASPD = newamt
-
-		/*	var/obj/item/bodypart/legr = get_bodypart(BODY_ZONE_R_LEG)
-			if(legr)
-				if(STASPD <= 10)
-					legr.sellprice = STASPD
-				if(STASPD > 10)
-					legr.sellprice = STASPD*2
-					if(STASPD > 12)
-						legr.sellprice = STASPD*4
-						if(STASPD > 15)
-							legr.sellprice = STASPD*6
-							if(STASPD > 17)
-								legr.sellprice += STASPD*8
-
-			var/obj/item/bodypart/legl = get_bodypart(BODY_ZONE_L_LEG)
-			if(legl)
-				if(STASPD <= 10)
-					legl.sellprice = STASPD
-				if(STASPD > 10)
-					legl.sellprice = STASPD*2
-					if(STASPD > 12)
-						legl.sellprice = STASPD*4
-						if(STASPD > 15)
-							legl.sellprice = STASPD*6
-							if(STASPD > 17)
-								legl.sellprice += STASPD*8	Commented out since no gating to fix goblins etc being farmed for truly ludicrous amounts*/
-
-			update_move_intent_slowdown()
-
-		if("fortune")
-			newamt = STALUC + amt
-			if(BUFLUC < 0)
-				BUFLUC = BUFLUC + amt
-				if(BUFLUC > 0)
-					newamt = STALUC + BUFLUC
-					BUFLUC = 0
-			if(BUFLUC > 0)
-				BUFLUC = BUFLUC + amt
-				if(BUFLUC < 0)
-					newamt = STALUC + BUFLUC
-					BUFLUC = 0
-			while(newamt < 1)
-				newamt++
-				BUFLUC--
-			while(newamt > 20)
-				newamt--
-				BUFLUC++
-			STALUC = newamt
-
-		/*	var/obj/item/organ/tongue/tongue = getorganslot(ORGAN_SLOT_TONGUE) //Superstition? also only really beggars and jesters can be farmed for this
-			if(tongue)
-				if(STALUC <= 10)
-					tongue.sellprice = STALUC
-				if(STALUC > 10)
-					tongue.sellprice = STALUC*2
-					if(STALUC > 12)
-						tongue.sellprice = STALUC*4
-						if(STALUC > 14)
-							tongue.sellprice = STALUC*6	Commented out since no gating to fix goblins etc being farmed for truly ludicrous amounts*/
-
-/proc/generic_stat_comparison(userstat as num, targetstat as num)
-	var/difference = userstat - targetstat
-	if(difference > 1 || difference < -1)
-		return difference * 10
+	switch(stat_key)
```

Comment:
this can be condensed into a single proc to avoid copy pasting the code 3 times by just having a return stats proc that returns a 2 length list with yours and the opponents stats